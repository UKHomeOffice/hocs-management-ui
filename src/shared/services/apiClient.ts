import axios from 'axios';

export default {
    createClient: () => {
        const client = axios.create();

        return {
            get: client.get,
            delete: client.delete,
            post: client.post,
            put: client.put
        };
    }
};
