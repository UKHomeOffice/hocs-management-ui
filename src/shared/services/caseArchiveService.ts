import axios from 'axios';
import ArchiveCaseModel from '../models/archiveCaseModel';

export const archiveCase = (archiveCaseModel: ArchiveCaseModel) => new Promise((resolve, reject) => axios
    .post('/api/case/archive', archiveCaseModel)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);
