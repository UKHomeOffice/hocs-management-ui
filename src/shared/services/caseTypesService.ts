import axios from 'axios';
import CaseType from '../models/caseType';
import Item from '../models/item';

const sortByLabel: ((a: Item, b: Item) => number) = (a, b) => {
    const labelA = a.label.toLocaleLowerCase();
    const labelB = b.label.toLocaleLowerCase();
    let comparison = 0;
    if (labelA > labelB) {
        comparison = 1;
    } else if (labelA < labelB) {
        comparison = -1;
    }
    return comparison;
};

export const getCaseTypes = () => new Promise<CaseType[]>((resolve, reject) =>
    axios.get('/api/case-types')
        .then((response) => {
            resolve(response.data.sort(sortByLabel));
        })
        .catch(error => reject(error))
);
