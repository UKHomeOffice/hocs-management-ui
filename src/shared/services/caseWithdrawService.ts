import axios from 'axios';
import WithdrawCaseModel from '../models/withdrawCaseModel';

export const withdrawCase = (withdrawCaseModel: WithdrawCaseModel) => new Promise((resolve, reject) => axios
    .post('/api/case/withdraw', withdrawCaseModel)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);
