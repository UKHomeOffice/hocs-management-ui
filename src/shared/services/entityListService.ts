import axios from 'axios';
import EntityListItem from 'shared/models/entityListItem';

export const getListItems = (listName: string) => new Promise<EntityListItem[]>((resolve, reject) => axios
    .get(`/api/entity/list/${listName}`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const createListItem = (item: EntityListItem, listName: string) => new Promise((resolve, reject) => axios
    .post(`/api/entity/list/${listName}`, item)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);

export const getItemDetails = (itemUUID: string) => new Promise<EntityListItem>((resolve, reject) => axios
    .get(`/api/entity/${itemUUID}`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const updateListItem = (item: EntityListItem, listName: string) => new Promise((resolve, reject) => axios
    .post(`/api/entity/list/update/${listName}`, item)
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);

export const deleteListItem = (uuid: string, listName: string) => new Promise((resolve, reject) => axios
    .delete(`/api/entity/list/${listName}`, { data: { entityUUID: uuid } })
    .then(() => resolve(null))
    .catch(reason => reject(reason))
);
