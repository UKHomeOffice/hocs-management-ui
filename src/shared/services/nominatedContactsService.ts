import axios from 'axios';
import NominatedContact from "../models/nominatedContact";

export const addNominatedContact = (nominatedContact: NominatedContact) => new Promise((resolve, reject) => axios
    .post('/api/nominated-contact', nominatedContact)
    .then(() => resolve())
    .catch(reason => reject(reason))
);
