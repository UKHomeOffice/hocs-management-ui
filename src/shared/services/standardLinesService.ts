import axios from 'axios';
import UpdateRequest from 'shared/pages/standardLineAmend/updateRequest';

export const addStandardLine = (standardLine: FormData) => new Promise((resolve, reject) => axios
    .post('/api/standard-lines', standardLine)
    .then(() => resolve())
    .catch(reason => reject(reason))
);

export const updateStandardLine = (standardLine: UpdateRequest) => new Promise((resolve, reject) => axios
    .post('/api/standard-lines/update', standardLine)
    .then(() => resolve())
    .catch(reason => reject(reason))
);

export const getAllStandardLines = () => new Promise((resolve, reject) => axios
    .get('/api/standard-lines/all')
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const getStandardLine = (standardLineUuid: string) => new Promise((resolve, reject) => axios
    .get(`/api/standard-lines/get/${standardLineUuid}`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);

export const expireStandardLine = (standardLineUuid: string) => new Promise((resolve, reject) => axios
    .post(`/api/standard-lines/expire/${standardLineUuid}`, {})
    .then(() => resolve())
    .catch(reason => reject(reason))
);

export const deleteStandardLine = (standardLineUuid: string) => new Promise((resolve, reject) => axios
    .delete(`/api/standard-lines/delete/${standardLineUuid}`, {})
    .then(() => resolve())
    .catch(reason => reject(reason))
);

export const downloadStandardLine = (standardLineUuid: string) => new Promise((resolve, reject) => axios
    .get(`/api/standard-lines/download/${standardLineUuid}`)
    .then(response => resolve(response.data))
    .catch(reason => reject(reason))
);
