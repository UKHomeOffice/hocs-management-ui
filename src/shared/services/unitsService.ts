import axios from 'axios';
import Unit from '../models/unit';
import Item from 'shared/models/item';

export const createUnit = (unit: Unit) => new Promise((resolve, reject) => axios
    .post('/api/units', unit)
    .then(() => resolve())
    .catch(reason => reject(reason))
);

export const getUnits = () => new Promise<[Item]>((resolve, reject) =>
    axios.get('/api/units')
        .then(response => resolve(response.data))
        .catch(error => reject(error))
);
