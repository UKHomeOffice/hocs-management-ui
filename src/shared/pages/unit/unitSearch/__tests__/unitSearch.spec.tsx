import React from 'react';
import { createBrowserHistory, History } from 'history';
import { act, wait, render, RenderResult } from '@testing-library/react';
import UnitSearch from '../unitSearch';
import * as UnitsService from '../../../../services/unitsService';
import { MemoryRouter } from 'react-router-dom';

let history: History<any>;

jest.mock('../../../../services/unitsService', () => ({
    __esModule: true,
    getUnits:jest.fn().mockReturnValue(Promise.resolve(
        [{
            displayName: 'Home Office General Property',
            shortCode: 'shortCode',
            value: '1aa9055d-0572-436b-a69d-4a97588f4ce4'
        }, {
            displayName: 'Home Office General Property',
            shortCode: 'shortCode',
            value: '1aa9055d-0572-436b-a69d-4a97588f4ce4'
        }]
    ))
}));

const getUnitsSpy = jest.spyOn(UnitsService, 'getUnits');
const dispatch = jest.fn();

const renderComponent = () => render(
    <MemoryRouter>
        <UnitSearch history={history}></UnitSearch>
    </MemoryRouter>
);

beforeEach(() => {
    history = createBrowserHistory();
    dispatch.mockReset();
});

describe('when the unitSearch component is mounted', () => {
    it('component should render', async () => {

        let wrapper: RenderResult;
        act(() => {
            wrapper = renderComponent();
        });

        await wait(() => {
            expect(getUnitsSpy).toHaveBeenCalled();
            expect(wrapper.container).toMatchSnapshot();
        });
    });
});