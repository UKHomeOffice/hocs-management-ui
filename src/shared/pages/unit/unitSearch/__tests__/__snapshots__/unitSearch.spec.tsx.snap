// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the unitSearch component is mounted component should render 1`] = `
<div>
  <div
    class="govuk-form-group"
  >
    <a
      class="govuk-back-link"
      href="/"
    >
      Back
    </a>
    <h1
      class="govuk-heading-xl"
    >
      Unit List
    </h1>
    <table
      class="govuk-table"
    >
      <caption
        class="govuk-table__caption"
      >
        Units
      </caption>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header"
            scope="row"
          >
            Home Office General Property
          </th>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header"
            scope="row"
          >
            Home Office General Property
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;